pymeasure/instruments/lecroy/lecroyT3DSO1204.py

```diff
@@ -0,0 +1,1192 @@
+        and the running status shows Stop.
+        Otherwise, the running state shows Ready, and the interface does not display the waveform.
+        • stopped : STOP is a part of the option of this command, but not a trigger mode of the
+        oscilloscope.""",
+        validator=strict_discrete_set,
+        values={"stopped": "STOP", "normal": "NORM", "single": "SINGLE", "auto": "AUTO"},
+        map_values=True
+    )
+
+    _trigger_select_values = {
+        "EDGE": [["C1", "C2", "C3", "C4", "LINE"], ["TI", "OFF"], [80e-9, 1.5]],
+        "DROP": [["C1", "C2", "C3", "C4"], ["TI"], [2e-9, 4.2]],
+        "GLIT": [["C1", "C2", "C3", "C4"], ["PS", "PL", "P2", "P1"], [2e-9, 4.2], [2e-9, 4.2]],
+        "RUNT": [["C1", "C2", "C3", "C4"], ["PS", "PL", "P2", "P1"], [2e-9, 4.2]],
+        "SLEW": [["C1", "C2", "C3", "C4"], ["IS", "IL", "I2", "I1"], [2e-9, 4.2]],
+        "INTV": [["C1", "C2", "C3", "C4"], ["IS", "IL", "I2", "I1"], [2e-9, 4.2]]
+    }
+
+    _trigger_select_short_command = "TRSE %s,SR,%s,HT,%s"
+    _trigger_select_normal_command = "TRSE %s,SR,%s,HT,%s,HV,%.2E"
+    _trigger_select_extended_command = "TRSE %s,SR,%s,HT,%s,HV,%.2E,HV2,%.2E"
+
+    _trigger_select = Instrument.control(
+        "TRSE?", _trigger_select_normal_command,
+        """ A string parameter that selects the condition that will trigger the acquisition of
+        waveforms.
+        Depending on the trigger type, additional parameters must be specified. These additional
+        parameters are grouped in pairs. The first in the pair names the variable to be modified,
+        while the second gives the new value to be assigned. Pairs may be given in any order and
+        restricted to those variables to be changed.
+        There are five parameters that can be specified. Parameters 1. 2. 3. are always mandatory.
+        Parameters 4. 5. are required only for certain combinations of the previous parameters.
+        1.  <trig_type>:={edge, slew, glit, intv, runt, drop}
+        2.  <source>:={c1, c2, c3, c4, line}
+        3.  - <hold_type>:={ti, off} for edge trigger.
+            - <hold_type>:={ti} for drop trigger.
+            - <hold_type>:={ps, pl, p2, p1} for glit/runt trigger.
+            - <hold_type>:={is, il, i2, i1} for slew/intv trigger.
+        4.  <hold_value1>:= a time value with unit.
+        5.  <hold_value2>:= a time value with unit.
+
+        Note:
+        • "line" can only be selected when the trigger type is "edge".
+        • If there is no unit(S/mS/uS/nS) added, it defaults to be S.
+        • The range of hold_values varies from trigger types. [80nS, 1.5S] for "edge" trigger,
+        and [2nS, 4.2S] for others.
+        """,
+        get_process=_trigger_select_get_process,
+        validator=_trigger_select_validator,
+        values=_trigger_select_values,
+        dynamic=True
+    )
+
+    @property
+    def trigger_select(self):
+        """ Refer to the self._trigger_select documentation. """
```

Comment:
Normally attributes starting with an underscore are "private". It is better to put the documentation to the public attribute (without leading underscore).

Only the property documentation is used, not that of the setter (method below).

Reply:
That makes sense.